### PR TITLE
Cleaning up Hora

### PR DESCRIPTION
**Pull request by wizardlyhel**
Cleaning up Hora

Status: **Ready for merge**

Reviewers: @ericmuyser @donnielrt @tedtate @hcstephencheung 
JIRA: https://mobify.atlassian.net/browse/CSOPS-1313
## Changes
- [x] remove any project specific hora actions
- [x] remove transaction as it is provided in ajs
- [x] provide more generic event actions
- [x] clean up tests
- [x] version bump
- [x] clean up readme (Wiki is separate)
- [x] pull in Eric's PR https://github.com/mobify/hora.js/pull/19
## Notes

Steve C wanted me to do this for awhile and it fell off my plate until CMS is needing to gather event action on carousel on a project.

### CODE REVIEW

**Comment by donnielrt** [comment]
@Helen-Mobify let's bump up the major number since we're breaking backwards-compatibility

File: src/js/hora.js [hunk 1]
```diff
@@ -41,6 +41,13 @@ define([
             }
         };
 
+        Hora.event = {
```
Comment (donnielrt) [inline src/js/hora.js]: [nitpick] We can add this in the initial Hora definition:

var Hora = {
    isDebug: false
    events: {
        // ...
    }
};

Comment (Helen-Mobify) [inline src/js/hora.js]: sure thing

File: src/js/hora.js [hunk 2]
```diff
@@ -585,90 +449,7 @@ define([
...
-         *      'price': '11.99',                 // Unit price.
-         *      'quantity': '1'                   // Quantity.
-         *     }
-         * ]);
-         *
-         * For more in-depth documentation on Google Analytics Ecommerce Tracking, please see:
-         * https://developers.google.com/analytics/devguides/collection/analyticsjs/ecommerce
-         *
-         */
-        Hora.transaction = {
```
Comment (donnielrt) [inline src/js/hora.js]: Do we not need this anymore?

Comment (Helen-Mobify) [inline src/js/hora.js]: Nope! AJS have [transaction](https://github.com/mobify/transaction) included

**Comment by Helen-Mobify** [comment]
@donnielrt Eyes please

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "type" : "git",
     "url" : "https://github.com/mobify/hora.js"
   },
-  "version": "1.1.0",
+  "version": "2.0.0",
```
Comment (donnielrt) [inline package.json]: We need to update bower.json too

File: src/js/hora.js [hunk 4]
```diff
@@ -335,196 +383,11 @@ define([
             }
         };
 
-        Hora.navigation = {
```
Comment (donnielrt) [inline src/js/hora.js]: We need to update the readme to remove these events

Comment (Helen-Mobify) [inline src/js/hora.js]: Yea. I'll remove the reference in readme... but how do we deal with the wiki ones?

File: dist/hora.js [hunk 5]
```diff
@@ -14,7 +14,13 @@ define([
         var $doc = $(document);
 
         var Hora = {
-            isDebug: false
+            isDebug: false,
+            event: {
```
Comment (donnielrt) [inline dist/hora.js]: If we're just using this locally (a user wouldn't need to call Hora.event.accordiong), we don't need to bind this to Hora, and could possibly just use a local variable `eventTitles`

Comment (Helen-Mobify) [inline dist/hora.js]: Yea, you are right. This is local to Hora. Let's not bind to Hora.

File: src/js/hora.js (hunk 4 above)
Comment (donnielrt) [inline src/js/hora.js]: Where's the wiki currently?

Comment (ericmuyser) [inline src/js/hora.js]: https://github.com/mobify/hora.js/wiki

You can edit it directly, or clone it (wiki's are git repos)

Comment (donnielrt) [inline src/js/hora.js]: Thanks @ericmuyser, @Helen-Mobify looks like the wiki is part of this same repo, so we can just edit it here

Comment (Helen-Mobify) [inline src/js/hora.js]: @ericmuyser Can we have 2 copies of it? Like Hora 1.x points to old wiki and Hora 2.x points to new wiki.

Comment (ericmuyser) [inline src/js/hora.js]: @Helen-Mobify Not that I know of, but you could make your changes and then add a link to an old revision. There's a list of revisions (for homepage) here: https://github.com/mobify/hora.js/wiki/Home/_history

Click on the SHA: https://github.com/mobify/hora.js/wiki/Home/beaccd754174b7bdaea27f1da19e45411d751138

**Comment by Helen-Mobify** [comment]
@ericmuyser Eyes please. I merged in your PR https://github.com/mobify/hora.js/pull/19

**Comment by ericmuyser** [comment]
I haven't tested this PR myself, but looking over it, it seems fine :)

**Comment by Helen-Mobify** [comment]
@donnielrt @ericmuyser I need +1s

I tested this on Carnival. Everything looks right. All events sent by Hora should have non-interaction parameter append to it.

**Comment by donnielrt** [comment]
@Helen-Mobify @ericmuyser doesn't like we need the `graphs` folder in there, seems like it can be added to the .gitignore and removed from repo?

**Comment by Helen-Mobify** [comment]
... I don't know what `graphs` fold do. @ericmuyser ?

**Comment by ericmuyser** [comment]
@donnielrt Good eye. Yes please delete it. Don't need to add it to gitignore. I was experimenting with flowhub, a cool visual programming tool, and it requires pushing a config file to the repo.

**Comment by donnielrt** [comment]
@Helen-Mobify 

one last non-blocking question, and then +1 from me - you've mentioned that we always want to set non-interaction to true, but looks like we're not doing that here: https://github.com/mobify/hora.js/pull/21/files#diff-0f019d99eeda68eef540d81cb6ade789R267. I'm assuming that our defaults mean that this is still set to true? 

On the other hand, we do set it manually at times: https://github.com/mobify/hora.js/pull/21/files#diff-0f019d99eeda68eef540d81cb6ade789R358. Do we need that here?

**Comment by Helen-Mobify** [comment]
@donnielrt Ahh those `Load` function's non interaction are leftover from last version. I'll clean those up too.

**Comment by donnielrt** [comment]
Cool, sounds good!

**Comment by Helen-Mobify** [comment]
@donnielrt Good eye

**Comment by Helen-Mobify** [comment]
Alright! I am merging @donnielrt @ericmuyser

**Pull request merged**